src/components/your_grants/manage_grant/modals/modalContentMilestoneDone.tsx

```diff
@@ -1,17 +1,77 @@
 import {
-  ModalBody, Flex, Text, Button, Box, Image,
+  ModalBody, Flex, Text, Button, Box, Image, useToast,
 } from '@chakra-ui/react';
-import React, { useState } from 'react';
+import config from 'src/constants/config';
+import { ApiClientsContext } from 'pages/_app';
+import React, { useContext, useState } from 'react';
+import { ApplicationMilestone } from 'src/graphql/queries';
+import ApplicationRegistryAbi from 'src/contracts/abi/ApplicationRegistryAbi.json';
+import { getFormattedDateFromUnixTimestampWithYear, getMilestoneMetadata } from 'src/utils/formattingUtils';
+import { useContract, useSigner } from 'wagmi';
 import MultiLineInput from '../../../ui/forms/multiLineInput';
 
 interface Props {
-  onClose: () => void;
+  milestone: ApplicationMilestone | undefined
+  done: () => void
 }
 
-function ModalContent({ onClose }: Props) {
+function ModalContent({ milestone, done }: Props) {
+  const { subgraphClient, validatorApi, workspaceId } = useContext(ApiClientsContext)!;
+  const [signerStates] = useSigner();
+  const applicationRegContract = useContract({
+    addressOrName: config.ApplicationRegistryAddress,
+    contractInterface: ApplicationRegistryAbi,
+    signerOrProvider: signerStates.data,
+  });
+  const toast = useToast();
+
   const [details, setDetails] = useState('');
   const [detailsError, setDetailsError] = useState(false);
 
+  const [isLoading, setIsLoading] = useState(false);
+
+  const markAsDone = async () => {
+    setIsLoading(true);
+
+    try {
+      const { data } = await validatorApi.validateApplicationMilestoneUpdate({ text: details });
+      console.log(`uploaded milestone update data to IPFS: ${data.ipfsHash}`);
+
+      const { milestoneIndex, applicationId } = getMilestoneMetadata(milestone)!;
+      // contract interaction
+      const transaction = await applicationRegContract.approveMilestone(
+        applicationId,
+        Number(milestoneIndex),
+        Number(workspaceId),
+        data.ipfsHash,
+        0,
+        config.blankAddress,
+        0,
```

Comment:
@adiwajshing `approveMilestone` function definition is changed after this PR was raised, please update as per implementation here - https://github.com/questbook/contracts/blob/main/contracts/ApplicationRegistry.sol#L266